dvc/remote/index.py

```diff
@@ -0,0 +1,257 @@
+import logging
+import os
+import struct
+import threading
+import zlib
+from binascii import unhexlify
+
+from funcy import chunks, concat, split, split_at
+
+from dvc.exceptions import DvcException
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_PROTOCOL = 1
+SUPPORTED_PROTOCOLS = [1]
+
+# Index format (v1) is the following:
+#
+# (all integer types are little-endian)
+#
+# Header
+# ------
+#   protocol_version (32-bit uint): index file protocol version
+#   dir_checksum_count (64-bit uint): number of .dir checksums in data section
+#   file_checksum_count (64-bit uint): number of file checksums in data section
+#   compressed_len (64-bit uint): length of data section
+#   crc - 32-bit CRC32 checksum of uncompressed data
+#
+# Data (zlib compressed)
+# ----------------------
+#   array of <dir_checksum_count> 128-bit MD5 .dir checksums
+#   array of <file_checksum_count> 128-bit MD5 file checksums
+_header_v1 = struct.Struct("<IQQ")
+_compress_crc_v1 = struct.Struct("<QI")
+
+
+def _verify_protocol(protocol=None):
+    if protocol is None:
+        protocol = DEFAULT_PROTOCOL
+    if protocol not in SUPPORTED_PROTOCOLS:
+        raise DvcException(
+            "unsupported remote index protocol version: {}".format(protocol)
```

Comment:
```suggestion
            "unsupported remote index protocol version: '{}'".format(protocol)
```
AFAIR we use `'` around most dynamic output.
There's a few more instances of this.